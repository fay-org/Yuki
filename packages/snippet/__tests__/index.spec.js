import Snippet from '../index'
import { mount } from '@vue/test-utils'

describe('Snippet', () => {
  it('should be render as element', () => {
    const snippet = mount(<Snippet text="In Test" />)
    expect(snippet.html()).toMatchSnapshot()
    expect(() => snippet.unmount()).not.toThrow()
  })
  it('should support different types', () => {
    const wrapper = mount({
      setup() {
        return { types: ['default', 'success', 'warning', 'error'] }
      },
      render() {
        return (
          <>
            {this.types.map((type, idx) => (
              <Snippet text={type} type={type} key={idx} />
            ))}
          </>
        )
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.default')).toBeTruthy()
    expect(wrapper.find('.success')).toBeTruthy()
    expect(wrapper.find('.warning')).toBeTruthy()
    expect(wrapper.find('.error')).toBeTruthy()
  })

  it('should support custom symbol ', () => {
    const wrapper = mount(<Snippet symbol="custom" text="inTest" />)
    expect(wrapper.find('span').text()).toBe('custominTest')
  })
  it('should support no copy', () => {
    const wrapper = mount(<Snippet copy="prevent" text="inTest" />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-snippet_copy').exists()).toBe(false)
  })
})
