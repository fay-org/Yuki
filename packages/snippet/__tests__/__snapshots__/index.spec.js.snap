// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet should be render as element 1`] = `
<div class="fect-snippet default"><span><span>$</span>In Test</span>
  <div class="fect-snippet_copy"><svg viewBox="0 0 24 24" width="22" height="22" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentcolor;">
      <path d="M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z"></path>
    </svg></div>
</div>
`;

exports[`Snippet should support different types 1`] = `
<div class="fect-snippet default"><span><span>$</span>default</span>
  <div class="fect-snippet_copy"><svg viewBox="0 0 24 24" width="22" height="22" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentcolor;">
      <path d="M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z"></path>
    </svg></div>
</div>
<div class="fect-snippet success"><span><span>$</span>success</span>
  <div class="fect-snippet_copy"><svg viewBox="0 0 24 24" width="22" height="22" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentcolor;">
      <path d="M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z"></path>
    </svg></div>
</div>
<div class="fect-snippet warning"><span><span>$</span>warning</span>
  <div class="fect-snippet_copy"><svg viewBox="0 0 24 24" width="22" height="22" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentcolor;">
      <path d="M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z"></path>
    </svg></div>
</div>
<div class="fect-snippet error"><span><span>$</span>error</span>
  <div class="fect-snippet_copy"><svg viewBox="0 0 24 24" width="22" height="22" stroke="currentColor" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" fill="none" shape-rendering="geometricPrecision" style="color: currentcolor;">
      <path d="M8 17.929H6c-1.105 0-2-.912-2-2.036V5.036C4 3.91 4.895 3 6 3h8c1.105 0 2 .911 2 2.036v1.866m-6 .17h8c1.105 0 2 .91 2 2.035v10.857C20 21.09 19.105 22 18 22h-8c-1.105 0-2-.911-2-2.036V9.107c0-1.124.895-2.036 2-2.036z"></path>
    </svg></div>
</div>
`;

exports[`Snippet should support no copy 1`] = `
<div class="fect-snippet default disabled"><span><span>$</span>inTest</span>
  <!---->
</div>
`;
